cylc/uiserver/handlers.py

```diff
@@ -192,6 +192,10 @@ def check_xsrf_cookie(self):
     def get(self, path):
         # authenticate the static handler
         # this provides us with login redirection and token caching
+        # accessing xsrf_token ensures xsrf cookie is set if it needs to be,
+        # e.g. setting it during request for /index.html
+        # to be available for next request to /userprofile
+        self.xsrf_token  # noqa
```

Comment:
Ah, brilliant!

Reply:
> That's _almost_ got it, however I think when the `index.html` is cached by the browser, the GET never goes through our static handler and so the XSRF cookie is not set 😢. This seems more prevalent on Firefox than Chrome

I think setting [`Cache-Control: no-cache`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#response_directives) does the trick 🤞

```suggestion
        if not path:
            # Request for /index.html
            # Accessing xsrf_token ensures xsrf cookie is set
            # to be available for next request to /userprofile
            self.xsrf_token
            # Ensure request goes through this method even when cached so
            # that the xsrf cookie is set on new browser sessions
            # (doesn't prevent browser storing the response):
            self.set_header('Cache-Control', 'no-cache')
```